File: internal/store/destino.go
package store

// Destino é um canal de publicação: um grupo de Telegram, um número de
// WhatsApp, etc. É o que a usuária cadastra no dashboard.
type Destino struct {
	ID     string `json:"id"`     // slug único (ex.: "ofertas-beleza")
	Nome   string `json:"nome"`   // nome amigável
	Tipo   string `json:"tipo"`   // "telegram" | "whatsapp" | ...
	Config string `json:"config"` // dado específico do provedor (chat_id, telefone, ...)
	Ativo  bool   `json:"ativo"`
}

File: src/Garimpei.Domain/Entities/Destino.cs
using Garimpei.Domain.Interfaces;

namespace Garimpei.Domain.Entities;

/// <summary>
/// Canal de publicação (grupo Telegram, número WhatsApp, etc.).
/// </summary>
public sealed class Destino : IOwnedEntity
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public required string Nome { get; init; }
    public required string Tipo { get; init; } // "telegram" | "whatsapp"
    public string Config { get; set; } = string.Empty; // chat_id, telefone, etc.
    public bool Ativo { get; set; } = true;
    public string OwnerUid { get; set; } = string.Empty;
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
